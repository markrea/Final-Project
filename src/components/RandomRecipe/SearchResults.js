import React from "react";
import PropTypes from "prop-types";
import "../../styles/SearchResults.css";

const Result = ({ results }) => {
  return (
    <div key={results.image} className="innerSearchResult">
      <img src={results.image} alt="mockAlt" className="card-photo" />
      <figcaption className="caption">{results.title}</figcaption>
      <a href={results.link} target="_blank" rel="noopener noreferrer">
        View Recipe
      </a>
    </div>
  );
};
const SearchResults = ({ searchResults }) => {
  return (
    <div className="SearchResults" data-testid="results-div">
      <div className="test">
        {searchResults.map((results) => (
          <Result key={results.id} results={results} />
        ))}
      </div>
    </div>
  );
};

Result.propTypes = {
  results: PropTypes.objectOf.isRequired,
};
SearchResults.propTypes = {
  searchResults: PropTypes.arrayOf.isRequired,
};

export default SearchResults;
