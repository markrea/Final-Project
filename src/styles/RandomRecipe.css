.randomButton{
    display:table-cell;

    border-radius: 10px 10px;
    margin: auto;   
    
    margin-top: 40px;
    width: 100px;    
    align-content: center;    
}
.randomResults{    
   
    margin: 0;
    padding: 0;
    text-align: center;   
    
}
