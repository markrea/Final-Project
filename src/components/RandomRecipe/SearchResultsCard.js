import React from "react";
import Proptypes from "prop-types";
import SearchResults from "./SearchResults";
import "../../styles/App.css";

const SearchResultsCard = ({ searchResults }) => {
  return (
    <div>
      <div className="search-results" data-testid="search-div">
        {searchResults && <SearchResults searchResults={searchResults} />}
      </div>
    </div>
  );
};
SearchResultsCard.propTypes = {
  searchResults: Proptypes.objectOf.isRequired,
};

export default SearchResultsCard;
