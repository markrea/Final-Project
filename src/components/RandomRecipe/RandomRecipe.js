import React, { useState } from "react";
import RandomSearch from "./RandomSearch";
import SearchResultsCard from "./SearchResultsCard";
import "../../styles/SearchByIngredients.css";
import "../../styles/RandomRecipe.css";

const RandomRecipe = () => {
  const [searchResults, setSearchResults] = useState(null);

  return (
    <div className="SearchByIngredients" data-testid="SearchByIngredients">
      <div className="SearchByIngredientsText">
        <h2>Random Recipe</h2>
        <div className="FormText">
          <div className="mainText">
            <h4>Can&apos;t decide what to make? See a random recipe!</h4>
          </div>
        </div>

        <RandomSearch setSearchResults={setSearchResults} />
        <div className="randomResults">
          <SearchResultsCard searchResults={searchResults} />
        </div>
      </div>
    </div>
  );
};

export default RandomRecipe;
