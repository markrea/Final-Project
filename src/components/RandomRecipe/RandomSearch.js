import React, { useState } from "react";
import Loader from "react-loader-spinner";
import "react-loader-spinner/dist/loader/css/react-spinner-loader.css";

import getRandomRecipe from "../../requests/getRandomRecipe";

import "../../styles/Search.css";
import "../../styles/RandomRecipe.css";

const RandomSearch = ({ setSearchResults }) => {
  const [loading, setLoading] = useState(false);


  const handleClick = async (event) => {
    setLoading(true);
    event.preventDefault();
    setSearchResults(await getRandomRecipe());
    setLoading(false);
  };

  return (
    <div className="RandomRecipe" data-testid="searchFormDiv">
      <button className="randomButton" type="button" onClick={handleClick}>
        Go!
      </button>

      {loading && (
        <Loader
          className="loader"
          type="ThreeDots"
          color="green"
          height={60}
          width={60}
          timeout={3000}
        />
      )}

    </div>
  );
};
export default RandomSearch;
