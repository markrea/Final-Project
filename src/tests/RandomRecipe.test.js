import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import "@testing-library/jest-dom/extend-expect";
import RandomRecipe from "../components/RandomRecipe/RandomRecipe";

describe("RandomRecipe", () => {
  it("components renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <RandomRecipe />
      </MemoryRouter>
    );
    expect(asFragment).toMatchSnapshot();
  });
  it("returns a SearchByIngredients div", () => {
    const { getByTestId } = render(
      <MemoryRouter>
        <RandomRecipe />
      </MemoryRouter>
    );
    expect(getByTestId("SearchByIngredients")).toBeInTheDocument();
    expect(getByTestId("SearchByIngredients")).toHaveClass(
      "SearchByIngredients"
    );
  });
  it("renders a title", () => {
    const { getByText } = render(
      <MemoryRouter>
        <RandomRecipe />
      </MemoryRouter>
    );
    expect(getByText("Random Recipe")).toBeInTheDocument();
  });
});
