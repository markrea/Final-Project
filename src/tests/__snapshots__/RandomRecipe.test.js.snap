// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RandomRecipe components renders correctly 1`] = `[Function]`;
